packages/main/src/MessageStrip.ts

```diff
@@ -155,6 +155,12 @@ class MessageStrip extends UI5Element {
 		this.fireEvent("close");
 	}
 
+	onBeforeRendering() {
+		if (this.designClasses === DesignClassesMapping.ColorSet1 || this.designClasses === DesignClassesMapping.ColorSet2) {
+			this.hideIcon = this.icon.length === 0;
```

Comment:
`hideIcon` is a public property and we can't just internally change its value, we can add internal property like _hideIcon, effectiveHideIcon or similar that considers both the public hideIcon value and the internal rules for showing/hiding an icon.

Reply:
Or, even simpler - with a getter like this:

```ts
get shouldHideIcon() {
   if (this.hideIcon) {
     return  true
   }

   if (this.designClasses === DesignClassesMapping.ColorSet1 || this.designClasses === DesignClassesMapping.ColorSet2) {
      return this.icon.length === 0;
   }
}
```
* it's just for the sake of the example - it can be written in more simplified way

Then, we can use the shouldHideIcon in the template - {{unless shouldHideIcon}} render the icon {{/unless}}

Reply:
Thanks, I will fix accordingly!

Reply:
Done